perllib/FixMyStreet/App/Controller/Hercules.pm

```diff
@@ -494,6 +494,51 @@ sub setup_categories_and_bodies : Private {
     @$contacts = grep { grep { $_ eq 'Waste' } @{$_->groups} } @$contacts;
 }
 
+sub receive_echo_event_notification : Path('/waste/echo') : Args(0) {
+    my ($self, $c) = @_;
+    $c->response->header(Content_Type => 'text/xml');
+
+    require SOAP::Lite;
+
+    $c->detach('soap_error', [ 'POST' ]) unless $c->req->method eq 'POST';
```

Comment:
Should we also check the SOAP action somewhere and bail if it's not one we expect?

Reply:
I wanted to use SOAP::Transport::HTTP (which does checks like this) but then it wants to generate the output and captures the $c->detaches and so on. So, yes, will do.